core/test/junit/saros/util/PathUtilsTest.java

```diff
@@ -0,0 +1,217 @@
+        expectedResult,
+        shortenedPath);
+  }
+
+  @Test
+  public void testRemoveLastSegmentsNone() {
+    Path path = Paths.get("foo", "bar");
+    Path shortenedPath = PathUtils.removeLastSegments(path, 0);
+
+    Path expectedResult = Paths.get("foo", "bar");
+
+    assertEquals(
+        "Dropping first segments did not result in expected outcome",
+        expectedResult,
+        shortenedPath);
+  }
+
+  @Test
+  public void testRemoveLastSegmentsSome() {
+    Path path = Paths.get("foo", "bar");
+    Path shortenedPath = PathUtils.removeLastSegments(path, 1);
+
+    Path expectedResult = Paths.get("foo");
+
+    assertEquals(
+        "Dropping first segments did not result in expected outcome",
+        expectedResult,
+        shortenedPath);
+  }
+
+  @Test
+  public void testRemoveLastSegmentsAll() {
+    Path path = Paths.get("foo", "bar");
+    Path shortenedPath = PathUtils.removeLastSegments(path, 2);
+
+    Path expectedResult = Paths.get("");
+
+    assertEquals(
+        "Dropping first segments did not result in expected outcome",
+        expectedResult,
+        shortenedPath);
+  }
+
+  @Test
+  public void testRemoveLastSegmentsMoreThanAll() {
+    Path path = Paths.get("foo", "bar");
+    Path shortenedPath = PathUtils.removeLastSegments(path, 3);
+
+    Path expectedResult = Paths.get("");
+
+    assertEquals(
+        "Dropping first segments did not result in expected outcome",
+        expectedResult,
+        shortenedPath);
+  }
+
+  @Test
+  public void testNormalizeResolvedParentReference() {
+    Path path = Paths.get("foo", "stuff", "..", "bar");
+    Path normalizedPath = PathUtils.normalize(path);
+
+    Path expectedResult = Paths.get("foo", "bar");
+
+    assertEquals(
+        "Normalizing path did not result in expected outcome", expectedResult, normalizedPath);
+  }
+
+  @Test
+  public void testNormalizeLocalReference() {
+    Path path = Paths.get(".");
+    Path normalizedPath = PathUtils.normalize(path);
+
+    Path expectedResult = Paths.get("");
+
+    assertEquals(
+        "Normalizing path did not result in expected outcome", expectedResult, normalizedPath);
+  }
+
+  @Test
+  public void testNormalizeMixed() {
+    Path path = Paths.get(".", "foo", ".", "", "stuff", "..", ".", "bar", ".", ".");
+    Path normalizedPath = PathUtils.normalize(path);
+
+    Path expectedResult = Paths.get("foo", "bar");
+
+    assertEquals(
+        "Normalizing path did not result in expected outcome", expectedResult, normalizedPath);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testNormalizeAbsolute() {
+    Path path = Paths.get("foo", "bar").toAbsolutePath();
+    PathUtils.normalize(path);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testNormalizeDanglingParentReference() {
+    Path path = Paths.get("..");
+    System.out.println("Path: '" + PathUtils.normalize(path) + "'");
```

Comment:
Assert.fail(...) ?

Reply:
The test fails anyways due to the missing exception. I just added to output to make debugging easier. These tests will have to be adjusted once we update to jUnit 4.13 anyway to use `assertThrows(...)` which allows specifying a message. See TODO at l.11.